apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ForeignKeySelector/ForeignKeySelector.tsx

```diff
@@ -284,42 +297,87 @@ export const ForeignKeySelector = ({
-                  </div>
-                  <div className="col-span-4 text-xs text-foreground-lighter text-right">
-                    {fk.schema}.{fk.table}
-                  </div>
-                  {fk.columns.length === 0 && (
-                    <Alert_Shadcn_ className="col-span-10 py-2 px-3">
-                      <AlertDescription_Shadcn_>
-                        There are no foreign key relations between the tables
-                      </AlertDescription_Shadcn_>
-                    </Alert_Shadcn_>
-                  )}
-                  {fk.columns.map((_, idx) => (
-                    <Fragment key={`${uuidv4()}`}>
-                      <div className="col-span-4">
-                        <Listbox
-                          id="column"
-                          value={fk.columns[idx].source}
-                          onChange={(value: string) => updateSelectedColumn(idx, 'source', value)}
-                        >
-                          <Listbox.Option key="empty" value={''} label="---" className="!w-[170px]">
-                            ---
-                          </Listbox.Option>
-                          {(table?.columns ?? [])
-                            .filter((x) => x.name.length !== 0)
-                            .map((column) => (
+              {isLoadingSelectedTable ? (
+                <div className="flex py-6 flex-col items-center justify-center space-y-2">
+                  <Loader2 size={14} className="animate-spin" />
+                  <p className="text-sm text-foreground-light">Loading table columns</p>
+                </div>
+              ) : (
+                <div className="flex flex-col gap-y-3">
+                  <label className="text-foreground-light text-sm">
+                    Select columns from{' '}
+                    <code className="text-xs">
+                      {fk.schema}.{fk.table}
+                    </code>{' '}
+                    to reference to
+                  </label>
+                  <div className="grid grid-cols-10 gap-y-2">
+                    <div className="col-span-5 text-xs text-foreground-lighter">
+                      {selectedSchema}.{table.name.length > 0 ? table.name : '[unnamed table]'}
+                    </div>
+                    <div className="col-span-4 text-xs text-foreground-lighter text-right">
+                      {fk.schema}.{fk.table}
+                    </div>
+                    {fk.columns.length === 0 && (
+                      <Alert_Shadcn_ className="col-span-10 py-2 px-3">
+                        <AlertDescription_Shadcn_>
+                          There are no foreign key relations between the tables
+                        </AlertDescription_Shadcn_>
+                      </Alert_Shadcn_>
+                    )}
+                    {fk.columns.map((_, idx) => (
+                      <Fragment key={`${uuidv4()}`}>
```

Comment:
Won't this unmount and remount every fragment on every render? I think keying on `${fk.schema}-${fk.table}-${idx}` should be fine here, as far as I can tell there's no way to rearrange the list

Reply:
omg yes! Good catch 🐐